### PR TITLE
Src: Fix Coverity Log Issues

### PR DESCRIPTION
**Pull request by zachdworkin**
Add locking to potentially unsafe log_fid.ops functions

### CODE REVIEW

**Comment by shijin-aws** [comment]
@zachdworkin efa fix is already in https://github.com/ofiwg/libfabric/pull/9849

**Comment by zachdworkin** [comment]
@shijin-aws Thanks, I missed that PR. I removed the patch that had that change in it.

File: src/log.c [hunk 1]
```diff
@@ -323,11 +323,15 @@ int DEFAULT_SYMVER_PRE(fi_log_enabled)(const struct fi_provider *prov,
...
 	uint64_t flags = 0;
+	int ret = 0;
 
 	if (ofi_prov_ctx(prov)->disable_logging)
 		flags |= FI_LOG_PROV_FILTERED;
 
-	return log_fid.ops->enabled(prov, level, subsys, flags);
+	pthread_mutex_lock(&common_locks.ini_lock);
+	ret = log_fid.ops->enabled(prov, level, subsys, flags);
+	pthread_mutex_unlock(&common_locks.ini_lock);
```
Comment (shefty) [inline src/log.c:333]: This introduces significant overhead to logging, even when it's disabled.  We don't want any of these locks in the logging calls

Comment (zachdworkin) [reply src/log.c:333]: Thanks! I was noticing that so I added the evaluating label. Do you have a different fix suggestion?

**Pull request closed without merging**